### PR TITLE
A0-4035: feature control pallet

### PR DESCRIPTION
**Pull request by pmikolajczyk41**
# Description

We include a new pallet: `FeatureControl`. Its purpose is explained in the Jira ticket. This PR brings the pallet implementation (with unit tests and benchmarks) and incorporates it to the runtime. There will be a direct follow-up PR for:
 - removing `liminal` feature flag from our repository
 - including ZK-related crates in the Rust workspace
 - hiding ZK features using the new pallet

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- I have added tests
- I have made neccessary updates to the Infrastructure
- I have created new documentation

### CODE REVIEW

File: pallets/feature-control/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+pub enum Feature {
+    /// The on-chain verifier feature involves:
+    /// - VkStorage pallet (for storing verification keys)
+    /// - smart contract chain extension exposing `verify` function
+    /// - SnarkVerifier runtime interface
+    #[codec(index = 0)]
+    OnChainVerifier,
+}
+
+/// The current storage version.
```
Comment (kostekIV) [inline pallets/feature-control/src/lib.rs:35]: nit: this is not public and to be honest doc is redundant

File: pallets/feature-control/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    use super::{weights::WeightInfo, *};
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// Item required for emitting events.
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+        /// Weight information for the pallet's extrinsics.
+        type WeightInfo: WeightInfo;
+        /// The origin that can modify the feature map.
+        type Controller: EnsureOrigin<Self::RuntimeOrigin>;
```
Comment (kostekIV) [inline pallets/feature-control/src/lib.rs:52]: nit: Controller may be a little confusing given we had controllers in the staking pallet.

**Review by kostekIV — APPROVED** [review]
lgtm, 

You can also add pub function:
* Pallet::is_feature_enabled(feature: Feature)

File: pallets/feature-control/src/lib.rs (hunk 1 above)
Comment (pmikolajczyk41) [reply pallets/feature-control/src/lib.rs]: right, I copied it without much consideration from another pallet; doc removed

**Review by DamianStraszak — APPROVED** [review]

**Pull request merged**